app/src/common/chromium/com/igalia/wolvic/browser/api/impl/YoutubeUrlHelper.java

```diff
@@ -0,0 +1,51 @@
+package com.igalia.wolvic.browser.api.impl;
+
+import org.chromium.url.GURL;
+
+import java.util.ArrayList;
+
+/**
+ * In addition to desktop UA YouTube requires the query parameter "app=desktop" to display 360
+ * videos correctly. This helper class implements the method to add this parameter to URL.
+ */
+public class YoutubeUrlHelper {
+    public static GURL maybeRewriteYoutubeURL(GURL url) {
+        if (!url.domainIs("youtube.com") && !url.domainIs("youtube-nocookie.com")) {
+            return url;
+        }
+
+        if (!url.getPath().equals("/watch")) {
+            return url;
+        }
+        
+        // Java version of GURL doesn't expose the same ReplaceComponents functionality as the
+        // native version, so we have to use string concatenation here.
+        return new GURL(url.getScheme() + "://" + url.getHost() + url.getPath() + "?" +
+                ensureAppIsSetToDesktop(url.getQuery()));
+    }
+
+    private static String ensureAppIsSetToDesktop(String query) {
```

Comment:
I think you can also use [Android's Uri](https://developer.android.com/reference/android/net/Uri.html) in this case, at least `getQueryParameter` seems convenient